distributed/shuffle/_merge.py

```diff
@@ -52,7 +52,7 @@ def _calculate_partitions(df: pd.DataFrame, index: IndexLabel, npartitions: int)
         meta=df._meta._constructor_sliced([0]),
         transform_divisions=False,
     )
-    df2 = df.assign(**{_HASH_COLUMN_NAME: partitions})
+    df2 = df.assign(**{_HASH_COLUMN_NAME: partitions.values})
```

Comment:
Does this need the same adjustment as we did in dask/dask?

Reply:
Good point, I'll port dask/dask#10462